doc/source/train/user-guides/experiment-tracking.rst

```diff
@@ -1,195 +1,364 @@
+        trainer.fit()
             
-            - `WandbLogger(id=UNIQUE_ID)`
-            - `CometLogger(experiment_key=UNIQUE_ID)`
-            - `MLFlowLogger(run_id=UNIQUE_ID)`
 
+Step 3: Log
+-----------
 
-        
-Automatic setup methods
-~~~~~~~~~~~~~~~~~~~~~~~
-Ray Train provides utilities for common experiment tracking libraries to automatically
-setup an experiment with the training run name and ID used by Ray Train. It also
-initializes the library in a way that only the rank 0 worker reports the results per
-default.
+You can customize within ``train_func`` how to log parameters, metrics, models, or media contents 
+just as you would with a single process training script. 
+You can also use native integrations that a particular tracking framework has with 
+specific training frameworks, for example ``mlflow.pytorch.autolog()``, 
+``lightning.pytorch.loggers.MLFlowLogger`` etc. 
+
+Step 4: Finish the run
+----------------------
+
+This step ensures that all logs are synced to the tracking service. Depending on the implementation of 
+various tracking libraries, sometimes logs are first stored locally before they are synced to the tracking 
+service. When training is finished, the node that the training worker resided on could be turned down by 
+ray autoscaler. Finishing the run makes sure that all logs are synced by the time training workers exit. 
+
+**Wandb**
+
+.. code-block:: python
+
+    # https://docs.wandb.ai/ref/python/finish
+    wandb.finish()
+
+**MLflow**
 
 .. code-block:: python
 
-    from ray import train
-    from ray.air.integrations.wandb import setup_wandb
+    # https://mlflow.org/docs/1.2.0/python_api/mlflow.html
+    mlflow.end_run()
+
+**Comet**
+
+.. code-block:: python
+
+    # https://www.comet.com/docs/v2/api-and-sdk/python-sdk/reference/Experiment/#experimentend
+    Experiment.end()
+
+Runnable Code
+=============
+
+PyTorch
+-------
+
+.. dropdown:: Log to Wandb (online) 
+
+    .. literalinclude:: ../../../../python/ray/train/examples/experiment_tracking//torch_exp_tracking_wandb.py
+            :emphasize-lines: 13, 14, 18, 19, 48, 49, 51, 52
+            :language: python
+            :start-after: __start__
+
+.. dropdown:: Log to file based MLflow (offline)         
+
+    .. literalinclude:: ../../../../python/ray/train/examples/experiment_tracking/torch_exp_tracking_mlflow.py
+        :emphasize-lines: 22, 23, 54, 55, 57, 58, 64
+        :language: python
+        :start-after: __start__
+        :end-before: __end__
+
+PyTorch Lightning
+-----------------
+
+The native Logger integration in PyTorch Lightning with W&B, CometML, MLFlow, 
+and Tensorboard can still be used seamlessly with Ray Train TorchTrainer.
+
+The following example will walk you through how. The code here is runnable. 
+There is a common shared piece of setting up a dummy model and dataloader
+just for demonstration purposes.
+        
+.. dropdown:: Define your model and dataloader (Dummy ones for demonestration purposes)
```

Comment:
Should we just include this in each collapsible block instead? I think as a user it's easier to just click open one block.